src/extension.ts

```diff
@@ -166,83 +168,54 @@ export function registerFileWatcher(projectProvider: ProjectProvider): void {
+            documentSelector: SUPPORTED_LANGUAGE_IDS,
+            synchronize: {
+                configurationSection: ["properties", "plaintext"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/bootstrap.properties"),
+                    workspace.createFileSystemWatcher("**/server.env")
+                ],
+            }
+        };
+
+        const serverOptions = prepareExecutable(LIBERTY_LS_JAR, requirements)
+
+        console.log("Creating new language client for Liberty Language Server");
+        libertyClient = new LanguageClient(LIBERTY_CLIENT_ID, localize("liberty.ls.output.dropdown"), serverOptions, clientOptions);
+        
+        const disposable = libertyClient.start();
+        context.subscriptions.push(disposable);
+
+        return libertyClient.onReady();
+    })
 }
 
-function startupJakartaLangServer(context: ExtensionContext) {
-    const jakartaLsPath = context.asAbsolutePath(path.join("jars", JAKARTA_LS_JAR));
-    // Language server options 
-    const serverOptions: ServerOptions = {
-        run: <Executable> { 
-            command: "java",
-            args: ["-jar", jakartaLsPath],
-            options: {stdio:"pipe"}
-        },
-        debug: <Executable> {
-            command: "java",
-            // TODO: using the debug arguments seems to still run the language server and open the debug port,
-            // but, the extension doesn't seem to work properly nor does it run the .onReady.then() commands
-            // const debugArgs = "-agentlib:jdwp=transport=dt_socket,server=y,suspend=n,address=1054";
-            // args: [debugArgs, "-jar", jakartaLsPath],
-            args: ["-jar", jakartaLsPath],
-            options: {stdio:"pipe"}
-        }
-    }
-
-    // Options to control the language client
-    const clientOptions: LanguageClientOptions = {
-        documentSelector: [{ scheme: "file", language: "java" }],
-        synchronize: {
-            configurationSection: ["java", "[java]"],
-            fileEvents: [
-                workspace.createFileSystemWatcher("**/*.java")
-            ],
-        }
-    };
-    console.log("Creating new language client for Jakarta Language Server")
-    jakartaClient = new LanguageClient(JAKARTA_CLIENT_ID, "Language Support for Jakarta", serverOptions, clientOptions);
-
-    context.subscriptions.push(jakartaClient.start());
-    return jakartaClient.onReady();
+function startupJakartaLangServer(context: ExtensionContext, api: JavaExtensionAPI) {
+    return resolveRequirements(api).then(requirements => {    
+        // Options to control the language client
+        const clientOptions: LanguageClientOptions = {
+            documentSelector: [{ scheme: "file", language: "java" }],
+            synchronize: {
+                configurationSection: ["java", "[java]"],
+                fileEvents: [
+                    workspace.createFileSystemWatcher("**/*.java")
```

Comment:
should we only watch for 'src'  folder?

Reply:
Personally, it makes sense to me to not restrict the lsp4jakarta server to a specific folder in a workspace